### PR TITLE
Polyhedron demo :Trying to fix the offset_meshing plugin

### LINKED ISSUE #4525: Demo: cannot stop offset meshing
When trying to stop a offset meshing in the demo, I get a crash of the demo with the message in the terminal: "libgcc_s.so.1 must be installed for pthread_cancel to work". It is strange since I can cancel a `Mesh_3` run.

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
Link with Threads::Treads to see if it fixes the weird error in #4525 

## Release Management

* Issue(s) solved (if any): fix #4525

### CODE REVIEW

File: Polyhedron/demo/Polyhedron/Plugins/Surface_mesh/CMakeLists.txt [hunk 1]
```diff
@@ -30,7 +30,7 @@ target_link_libraries(mesh_simplification_plugin PUBLIC scene_surface_mesh_item
 
 qt5_wrap_ui( remeshingUI_FILES  Remeshing_dialog.ui)
 polyhedron_demo_plugin(offset_meshing_plugin Offset_meshing_plugin ${remeshingUI_FILES})
-target_link_libraries(offset_meshing_plugin PUBLIC scene_surface_mesh_item scene_polygon_soup_item)
+target_link_libraries(offset_meshing_plugin PUBLIC scene_surface_mesh_item scene_polygon_soup_item Threads::Threads)
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Plugins/Surface_mesh/CMakeLists.txt]: You must add `find_package(Thread)` before.

Comment (lrineau) [reply Polyhedron/demo/Polyhedron/Plugins/Surface_mesh/CMakeLists.txt]: `Threads`, actually.

**Comment by maxGimeno** [comment]
Doesn't work

File: Polyhedron/demo/Polyhedron/Plugins/Surface_mesh/CMakeLists.txt [hunk 2]
```diff
@@ -28,9 +28,14 @@ qt5_wrap_ui( mesh_simplificationUI_FILES  Mesh_simplification_dialog.ui)
...
-qt5_wrap_ui( remeshingUI_FILES  Remeshing_dialog.ui)
-polyhedron_demo_plugin(offset_meshing_plugin Offset_meshing_plugin ${remeshingUI_FILES})
-target_link_libraries(offset_meshing_plugin PUBLIC scene_surface_mesh_item scene_polygon_soup_item Threads::Threads)
+find_package(Threads)
+if(Threads_FOUND)
+  qt5_wrap_ui( remeshingUI_FILES  Remeshing_dialog.ui)
+  polyhedron_demo_plugin(offset_meshing_plugin Offset_meshing_plugin ${remeshingUI_FILES})
+  target_link_libraries(offset_meshing_plugin PUBLIC scene_surface_mesh_item scene_polygon_soup_item Threads::Threads)
+else()
+  message("NOTICE : The Offset_meshing_plugin needs pthreads and will not be compiled")
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Plugins/Surface_mesh/CMakeLists.txt]: `pthreads` is only the threads library for POSIX systems. This message would be wrong on Windows.

**Pull request closed without merging**